### PR TITLE
Add a mushroom trader hardsuit

### PR DESCRIPTION
**Pull request by ShiftyRail**
Sprites by @otatoh / basil
Spawn in the trader's locker, along with their other hardsuits.

Has a toggle to dim incoming light, allowing you to walk in open sunlight, at the cost of reducing visibility (same overlay as a welding helmet).
As an aside, I fucking hate how this overlay is generated and I wish I hadn't looked at it.

:cl:
- rscadd: Mushroom trader hardsuits and helmets are now available for Vox traders. The helmet has an option to dim the incoming light.

### CODE REVIEW

**Comment by Lunstra** [comment]
based, roundstart trader mushbois when, speedmerge whence.

File: code/modules/maps/spawners/spawners.dm [hunk 1]
```diff
@@ -840,6 +840,10 @@
 			new /obj/item/clothing/suit/space/vox/civ/trader/stealth(src.loc) // black hardsuit. Not capable of any form of stealth systems or shit like that
 			new /obj/item/clothing/head/helmet/space/vox/civ/trader/stealth(src.loc)
 
+	// Mushie suits !
+	new /obj/item/clothing/suit/space/vox/civ/mushmen
```
Comment (ShiftyRail) [inline code/modules/maps/spawners/spawners.dm]: This lacks the (loc), hold on

**Comment by Kurfursten** [comment]
I approve of roundstart mushroom traders, and having a mushroom suit means we could have latejoin ones too

**Comment by Arthurdentist** [comment]
new buisness idea; what if some stations had mushroom traders instead of vox traders, for more distinct gameplay on each station

**Comment by Dacendeth** [comment]
A mushroom trader station where they get eggs to grow vox to get observers in the round again?

**Comment by Kurfursten** [comment]
Being a mushroom person is so intensely painful, I believe it should only be voluntary.

**Pull request merged**